### PR TITLE
Tool support section

### PR DESCRIPTION
**Pull request by allegroCoder**

### CODE REVIEW

File: main.tex [hunk 1]
```diff
@@ -432,6 +436,54 @@ \section{Using Domain-Specific Languages to improve the
...
+models, this can help the design for resiliency. Conditional partial order graphs (CPOG)
+(available in the Workcraft framework), for instance, can support the development of
+processor instruction sets. In Figure \ref{fig:screenshot}, all the steps from the
+specification of each instruction, to the synthesis of the final microcontroller are
+depicted. Instructions can be specified in the form of algebraic equations at first, then the
+graphs can be imported, encoded and synthesised in the form of a microcontroller. The latter
+can be mapped with the usage of a technology library and eventually simulated: internal and
+external signals can be visualised via digital waveforms, where dependencies among signals
+are shown. The development of a domain specific language (DSL), that can be used for the
+specification of the instructions, still remains a gap of this flow. Having a scalable, yet
```
Comment (snowleopard) [inline main.tex]: > still remains a gap of this flow

I would rephrase this in a more positive light: we are developing a DSL for bridging the gap between formal methods toolsets such Rodin and hardware synthesis toolsets such as Workcraft.

File: main.tex [hunk 2]
```diff
@@ -432,6 +436,54 @@ \section{Using Domain-Specific Languages to improve the
...
+	\label{fig:screenshot}
+\end{center}
+\end{figure*}
+
+The CPOG and Dataflow Structures (DFS) were used together to design an asynchronous
+reconfigurable pipeline for the computation of the ordinal pattern encoding. CPOG was used
+for the design of the control unit, which is used for the reconfigurability of the pipeline
+(3-18 stages). DFS supported the design of the datapath asynchronous components instead. Each
+module has indeed been modelled and simulated in Workcraft at first, and then translated into
+hardware description (VHDL). The DFS model matches the HDL description, the asynchronous 4-
```
Comment (snowleopard) [inline main.tex]: I believe Workcraft generates Verilog, not VHDL?

File: main.tex [hunk 3]
```diff
@@ -432,6 +436,54 @@ \section{Using Domain-Specific Languages to improve the
 without building an expensive prototype, and advanced types of hosting language
 encoding initial system specification provide verified-by-construction result. 
 
+\section{Tool support}
+
+Formalising the design specifications is a good approach to the development of a resilient
+system. Specifications, represented via mathematical models, can indeed be verified and
```
Comment (snowleopard) [inline main.tex]: > mathematical models

It may be worth sticking to the term `DSL` throughout the paper (instead of a `model`) to avoid confusion. There is not a very big conceptual difference, but we are trying to make a general point about the interplay of DSLs in Workcraft.

**Comment by snowleopard** [comment]
Thanks @allegroCoder! I've added a few minor comments.

File: main.tex (hunk 1 above)
Comment (allegroCoder) [reply main.tex]: Done!

File: main.tex (hunk 2 above)
Comment (allegroCoder) [reply main.tex]: I translated the model in VHDL manually. I would turn VHDL into Verilog, since Workcraft can generate verilog files.

File: main.tex (hunk 3 above)
Comment (allegroCoder) [reply main.tex]: Done!

**Comment by allegroCoder** [comment]
I pushed the corrections, thanks @snowleopard !

**Comment by snowleopard** [comment]
@allegroCoder Thanks, it's fine as a first draft. We might need to shorten this later.

I'll let @geo2a merge this if he is happy.

**Comment by allegroCoder** [comment]
Sounds good to me! I will stay tuned :)
